src/coreclr/vm/jitinterface.cpp

```diff
@@ -2855,6 +2889,52 @@ void CEEInfo::ScanTokenForDynamicScope(CORINFO_RESOLVED_TOKEN * pResolvedToken,
     ScanToken(pModule, pResolvedToken, th, pMD);
 }
 
+CORINFO_OBJECT_HANDLE CEEInfo::getJitHandleForObject(OBJECTREF obj)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    }
+    CONTRACTL_END;
+
+    if (m_pJitHandles == nullptr)
+    {
+        m_pJitHandles = new SArray<OBJECTHANDLE>();
+    }
+
+    OBJECTHANDLE handle;
+    GCPROTECT_BEGIN(obj);
+    handle = AppDomain::GetCurrentDomain()->CreateHandle(obj);
+
+    // We know that handle is aligned so we use the lowest bit as a marker
+    // "this is a handle, not a frozen object".
+    m_pJitHandles->Append(handle);
+    GCPROTECT_END();
```

Comment:
```suggestion
        OBJECTHANDLEHolder handle = AppDomain::GetCurrentDomain()->CreateHandle(obj);
    m_pJitHandles->Append(handle);
    handle.SuppressRelease();

    // We know that handle is aligned so we use the lowest bit as a marker
    // "this is a handle, not a frozen object".
```
`GCPROTECT_BEGIN` is not needed here.
We should make sure to release the handle in case `Append` fails.